bridgesync/downloader_test.go

```diff
@@ -326,28 +328,139 @@ func TestFindCall(t *testing.T) {
 	require.Equal(t, bridgeAddr, found.To)
 }
 
+func TestFindCallWithMixedMethods(t *testing.T) {
+	bridgeAddr := common.HexToAddress("0x10")
+	fromAddr := common.HexToAddress("0x20")
+	logger := logger.WithFields("module", "test")
+
+	// Test case: Transaction with mixed method calls
+	// First call: getProxiedTokensManager (unrecognized)
+	// Second call: claimAsset (recognized)
+	// Third call: claimMessage (recognized)
+	rootCall := call{
+		To:   common.HexToAddress("0x01"),
+		From: fromAddr,
+		Err:  nil,
+		Calls: []call{
+			{
+				To:    bridgeAddr,
+				From:  fromAddr,
+				Err:   nil,
+				Input: []byte{0x38, 0xb8, 0xfb, 0xbb}, // getProxiedTokensManager
+			},
+			{
+				To:    bridgeAddr,
+				From:  fromAddr,
+				Err:   nil,
+				Input: claimAssetEtrogMethodID, // claimAsset
+			},
+			{
+				To:    bridgeAddr,
+				From:  fromAddr,
+				Err:   nil,
+				Input: claimMessageEtrogMethodID, // claimMessage
+			},
+		},
+	}
+
+	// Test that findCall continues searching and finds the first valid claim method
+	found, err := findCall(rootCall, bridgeAddr, func(call call) (bool, error) {
+		// Simulate tryDecodeClaimCalldata behavior
+		if len(call.Input) < 4 {
+			return false, fmt.Errorf("input too short")
+		}
+		methodID := call.Input[:4]
+
+		if bytes.Equal(methodID, claimAssetEtrogMethodID) || bytes.Equal(methodID, claimMessageEtrogMethodID) {
+			return true, nil // Found a valid claim method
+		}
+
+		// Unrecognized method ID - return false, nil to continue searching
+		return false, nil
```

Comment:
WDYT, to reduce a couple of lines of code?
```suggestion
		isClaimInvoked := bytes.Equal(methodID, claimAssetEtrogMethodID) || bytes.Equal(methodID, claimMessageEtrogMethodID)

		return isClaimInvoked, nil
```